include/envoy/server/admin.h

```diff
@@ -15,6 +16,39 @@
 namespace Envoy {
 namespace Server {
 
+class AdminStream {
+public:
+  virtual ~AdminStream() {}
+
+  /**
+   * @param end_stream set to false for streaming response. Default is true, which will close
+   * connection on socket.
+   */
+  virtual void setEndStreamOnComplete(bool end_stream) PURE;
+
+  /**
+   * @param cb callback to be added to the list of callbacks invoked by onDestroy() when connection
+   * is dropped.
+   */
+  virtual void addOnDestroyCallback(std::function<void()> cb) PURE;
+
+  /**
+   * @return Http::StreamDecoderFilterCallbacks* to be used by the handler to get socket for data
```

Comment:
We are not really getting socket data here. We are getting HTTP request data, etc. I would rephrase.